src/sentry/static/sentry/app/components/charts/tableChart/index.jsx

```diff
@@ -0,0 +1,392 @@
+
+      if (!data) return [];
+
+      let totalRow = showColumnTotal
+        ? [
+            [
+              // Label for Total Row
+              columnTotalLabel,
+
+              // Need to fill empty columns between label and `dataStartIndex`,
+              ...[...Array(dataStartIndex - 1)].map(() => ''),
+
+              // totals for each data column
+              ...dataTotals.columnTotals,
+
+              // grand total if `showRowTotal` is enabled
+              ...(showRowTotal ? [dataTotals.total] : []),
+            ],
+          ]
+        : [];
+
+      return [...data, ...totalRow];
+    }
+
+    render() {
+      let {
+        className,
+        children,
+        data,
+        dataStartIndex,
+        getValue,
+        showRowTotal,
+        showColumnTotal,
+        shadeRowPercentage,
+        renderTableHeader,
+        renderBody,
+        widths,
+        ...props
+      } = this.props;
+
+      // If we need to calculate totals...
+      let dataTotals =
+        showRowTotal || showColumnTotal || shadeRowPercentage ? this.getTotals(data) : [];
+      let dataMaybeWithTotals = this.getDataWithTotals(dataTotals);
+
+      // For better render customization
+      let isRenderProp = typeof children === 'function';
+      let renderProps = {
+        data,
+        dataTotals,
+        dataMaybeWithTotals,
+        dataStartIndex,
+        getValue,
+        showRowTotal,
+        showColumnTotal,
+        shadeRowPercentage,
+        widths,
+        ...props,
+      };
+
+      return (
+        <Panel className={className}>
+          {isRenderProp ? (
+            children(renderProps)
+          ) : (
+            <React.Fragment>
+              {renderTableHeader(renderProps)}
+              {renderBody(renderProps)}
+            </React.Fragment>
+          )}
+        </Panel>
+      );
+    }
+  }
+)`
+  flex: 1;
+`;
+
+export default TableChart;
+
+export const TableChartRow = styled(
+  class extends React.Component {
+    static propTypes = {
+      /**
+       * Show percentage as a bar in the row
+       */
+      showBar: PropTypes.bool,
+      /**
+       * Total value of row
+       */
+      value: PropTypes.number,
+      /**
+       * Total value of all rows
+       */
+      total: PropTypes.number,
+    };
+
+    render() {
+      let {className, showBar, total, value, children} = this.props;
+      let barWidth =
+        total > 0 && typeof value === 'number' ? Math.round(value / total * 100) : 0;
+
+      return (
+        <PanelItem className={className}>
+          {children}
+          {showBar && <TableChartRowBar width={barWidth} />}
+        </PanelItem>
+      );
+    }
+  }
+)`
+  position: relative;
+  flex: 1;
+`;
+
+/**
+ * Shows relative percentage as width of bar inside of a table's row
+ */
+export const TableChartRowBar = styled(({width, ...props}) => <div {...props} />)`
+  position: absolute;
+  top: 0;
+  bottom: 0;
+  left: 0;
+  right: ${p => 100 - p.width}%;
```

Comment:
Would it have been better to set this as a inline style, so that we don't generate a new class for ever different width?

Reply:
Hmmm that's a good question, not sure.